### PR TITLE
fix for pypi underscore replace

### PR DESCRIPTION
**Pull request by declanhunter**
I noticed #23 and decided to fix it with @DarthHater 

I have adjusted the test to capture the issue and replace the code with the most broad solution

### CODE REVIEW

**Comment by DarthHater** [comment]
@pombredanne @stevespringett , this is @declanhunter first open source contribution (ever!), and he's @glenhunter from @sonatype 's kiddo, so go easy on him :)

Way to go Declan!

File: src/package-url.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ class PackageURL {
   }
 
   _handlePyPi() {
-    this.name = this.name.toLowerCase().replace('_', '-');
+    this.name = this.name.toLowerCase().replace(/_/g, '-');
```
Comment (DarthHater) [inline src/package-url.js:60]: Looking at this with @declanhunter , we discovered (and I had no clue about this!) that `replace` with a `substring` only replaces the first occurrence. `replaceAll` doesn't appear to be available in `node`, so this `regex` solution for `replace` seemed the most broadly acceptable solution

**Review by pombredanne — APPROVED** [review]
Hey! Thanks ++ for taking a shot at this. This is looking great!
@steven-esser ? any feedback?

**Review by steven-esser — APPROVED** [review]
@pombredanne looks good to me

**Pull request merged**